crates/noirc_errors/src/debug_info.rs

```diff
@@ -0,0 +1,39 @@
+use std::collections::HashMap;
+
+//use noirc_errors::Span;
+use serde::{Deserialize, Serialize};
+//use fm::FileId;
+use crate::Location;
+
+#[derive(Default, Debug, Clone, Deserialize, Serialize)]
+pub struct DebugInfo {
+    /// Map opcode index of an ACIR circuit into the source code location
+    pub locations: HashMap<usize, Location>,
+}
+
+impl DebugInfo {
+    pub fn new(locations: HashMap<usize, Location>) -> Self {
+        DebugInfo { locations }
+    }
+
+    pub fn update_acir(&mut self, opcode_idx: Vec<usize>) {
+        let mut new_locations = HashMap::new();
+        for (i, idx) in opcode_idx.iter().enumerate() {
+            if self.locations.contains_key(idx) {
+                new_locations.insert(i, self.locations[idx]);
+            }
+        }
+        self.locations = new_locations;
+    }
```

Comment:
So this is meant to filter `self.locations` to only those contained within `opcode_idx`?
Also what is `opcode_idx` meant to represent? Opcode indices into acir or something else? Can we change it to be plural as well: `opcode_indices` or `acir_opcode_indices` since this isn't the acir crate.

Reply:
Done, I have added some documentation for the function as well.